cot/src/auth.rs

```diff
@@ -152,11 +215,41 @@ pub trait User {
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 #[serde(untagged)]
 pub enum UserId {
+    /// A user ID that is an integer.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use cot::auth::UserId;
+    ///
+    /// let user_id = UserId::Int(42);
+    /// ```
     Int(i64),
+    /// A user ID that is a string.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use cot::auth::UserId;
+    ///
+    /// let user_id = UserId::String("42".to_string());
```

Comment:
`UserId` being String is probably when email or similar data is used for that purpose, let's point to that here as well
```suggestion
    /// let user_id = UserId::String("forty_two@example.com".to_string());
```